aws_xray_sdk/ext/sqlalchemy/util/decerators.py

```diff
@@ -46,7 +47,11 @@ def wrapper(*args, **kw):
                         sql = None
         if sql is not None:
             if getattr(c._local, 'entities', None) is not None:
-                subsegment = xray_recorder.begin_subsegment(sql['url'], namespace='remote')
+                # Strip URL of ? and following text
+                sub_name = strip_url(sql['url'])
+                # Ensure url has a valid characters. 
+                sub_name = ''.join([c for c in sub_name if c in _valid_name_characters])
```

Comment:
This is a duplicate to https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/entity.py#L33. IMO when we see a warning on invalid characters we fix from its source, like the `strip_url` here. But we should not deliberately skip the log warning by placing ` sub_name = ''.join([c for c in sub_name if c in _valid_name_characters])` everywhere. 

Otherwise it looks fine.

Reply:
Without that code, we would still end up logging warnings for certain SQL URLs. For example 'échéanciers' is a valid name or a database in postgres, but would be stripped to 'chanciers' for x-ray... I left this code  to prevent the warnings  as the sqlalchemy plugin is setting this, so it is not in users control.. If you want it removed I will, let me know.

Reply:
I'm fine with special casing the sql subsegment name. I will create a follow up issue for further write an utility module that strips unicode for patchers that might capture unicode as subsegment name.